Major refactor of clinvar_raw to simplify code flow

theferrit32 opened the pull request:
- Removes use of core.async channels and futures
- Converts some functions into pure functions with no side effects

theferrit32 committed:
Major refactor of clinvar_raw initiated. clinvar-raw.generate_local_topic/-main2 uses this, eliminating the async pieces of -main, and eliminates the middle step of uploading a drop message topic when generating clinvar-raw.

larrybabb commented:
@theferrit32 is there an issue on the github board to associate with this?

theferrit32 commented:
@larrybabb this was done as part of the work to ingest the larger test dataset. It significantly speeds things up, and makes it less error prone. I would like to clean this up before merging to master but it is not a high priority because it is not blocking anything else right now.

An issue could be added and put in the backlog.

theferrit32 committed:
Switch to clinvar-raw implementation without async. Remove async code.

theferrit32 committed:
Add testdata 20220523 archive. Contains a few CNVs

theferrit32 committed:
Remove async code from clinvar_raw.generate_local_topic

tbl3rd commented on src/clinvar_raw/core.clj:
```diff
@@ -12,23 +12,17 @@
+    (.start (Thread. (partial stream/process-drop-messages (cfg/app-config))))
+    (.start (Thread. (partial stream/send-producer-messages
+                              (cfg/app-config)
+                              (cfg/kafka-config (cfg/app-config)))))
```
Why not just delete this?

tbl3rd commented on src/clinvar_raw/stream.clj:
```diff
@@ -50,23 +45,25 @@
-(defn google-storage-line-reader [bucket filename]
+(defn google-storage-line-reader
   "Returns a reader to the storage object. Caller must open (with-open)"
+  [bucket filename]
```
The docstring could explain bucket and filename.

tbl3rd commented on src/clinvar_raw/stream.clj:
```diff
@@ -50,23 +45,25 @@
-(defn line-map-to-event [line-map entity-type release_date event-type]
+(defn line-map-to-event
   "Parses a single line of a drop file, transforms into an event object map"
+  [line-map entity-type release_date event-type]
```
Document the parameters?

tbl3rd commented on src/clinvar_raw/stream.clj:
```diff
@@ -133,127 +107,109 @@
+  "Same as select-keys, except only selects a given key k if the value of (get m k) is not nil"
+  [m keys]
+  (->> (select-keys m keys)
+       dissoc-nil-values))
```
Where is this used?

tbl3rd commented on src/clinvar_raw/stream.clj:
```diff
@@ -133,127 +107,109 @@
+(defn dissoc-nil-values
+  "Removes keys from maps for which the value is nil."
+  [input-map]
+  (apply dissoc input-map (filter #(= nil (get input-map %)) (keys input-map))))
```
```clojure
(defn dissoc-nil-values
  "Remove entries with nil values from `input` map."
  [input]
  (into {} (remove (comp nil? val) input)))
```

tbl3rd commented on src/clinvar_raw/stream.clj:
```diff
@@ -133,127 +107,109 @@
+               (log/info "Finished processing clinvar drop")
+               (log/info "Sending messages to output topic")
+               (doseq [output-m output-messages]
+                 (send-update-to-exchange producer output-topic output-m))))))))))
```
I left some comments on the /tbl/async-refactor/ branch.

theferrit32 committed:
Fixes #61. Process files from bucket one line at a time to avoid outofmemoryerror.

theferrit32 committed:
Comment out tests using older methods

theferrit32 committed:
Add functions to filter the clinvar-raw topic to a set of variants

theferrit32 committed:
Simplify cloudbuild to stop trying to cache layers

theferrit32 committed:
Set log filter to INFO. Increase poll-based consumer group timeout and reduce poll loop to 1 msg.

theferrit32 committed:
Reduce log noise. Increase poll wait time

theferrit32 committed:
Add 10 min backoff when poll returns no records. Increase max poll batch size.

theferrit32 committed:
Improve a log with select-keys-nested. Refactor obj-max to use reduce instead of loop.

theferrit32 committed:
Add apply-to-if and update select-keys-nested to only include key if value is not nil

theferrit32 committed:
Add to comments. Make stream_filter auto-close the consumer it opens

theferrit32 committed:
Merge branch 'main' into async-refactor

theferrit32 committed:
Add short test for select-keys-nested

theferrit32 committed:
Add nippy and genegraph's rocksdb module.

theferrit32 committed:
Adding deduplication functionality to clinvar-raw

theferrit32 committed:
Adding ability to include all of specific types in filtered output. And read from file instead of kafka.

theferrit32 committed:
Add functions to download/write/read gzip files. Fix create-db-path. Add dup filtration to clinvar-raw output stream.

theferrit32 committed:
Make clinvar-streams.config/app-config a function

theferrit32 committed:
Fix reference in clinvar-raw streaming dedup check.

larrybabb reviewed (approved):
As usual I will defer to the clojure experts for final say, but these refinements look sound to me. Pls don't delay in merging them if they are needed to get our urgent goals accomplished.

theferrit32 committed:
Bump clinvar-raw poll timeout to 12hrs.

theferrit32 commented:
I've addressed feedback from Tom aside from the file processing. That should be revisited later to more match other patterns for large sequence or file processing in our code.
https://github.com/clingen-data-model/clinvar-streams/issues/63
